### PR TITLE
chore: update yarn to current stable

### LINKED ISSUE #27: Update yarn to current stable
Dependency work in `aint` is more work than it should be, mostly because `yarn@1` has blunt dependency management tools. Let's update to `yarn@4` to get the benefit of a more recent `upgrade-interactive`

### PR DESCRIPTION
**Pull request by computerist**
Fixes #27

### CODE REVIEW

**Comment by causaly-mark** [comment]
Let's stick with yarn for now, @stathismor... if for no other reason than this was a pain in the backside 😆

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,13 +27,16 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [14.x]
+        node-version: [20.x]
 
     steps:
       - uses: actions/checkout@v2
```
Comment (stathismor) [inline .github/workflows/ci.yml:33]: nit: No need, but might as well upgrade these to v4.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -47,8 +50,10 @@ jobs:
           key: ${{ runner.os }}-build-${{ github.sha }}
 
       - name: Install dependencies
-        if: steps.yarn-cache.outputs.cache-hit != 'true'
-        run: yarn --ignore-scripts --frozen-lockfile
+        run: yarn --immutable
+
+      - name: Perform a yarn install
+        run: yarn install
```
Comment (stathismor) [inline .github/workflows/ci.yml:56]: question: Are you sure this is needed, when `yarn --immutable` is run just above?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -76,7 +84,16 @@ jobs:
       - uses: actions/cache@v2
         with:
           path: '**/build'
-          key: ${{ runner.os }}-build-${{ github.sha }}
+          key: ${{ runner.os }}-build-${{ github.sha }}      
+
+      - name: Install dependencies
```
Comment (stathismor) [inline .github/workflows/ci.yml:89]: I don't these steps are needed, as they are done in the previous job, `build`, that this one needs. Same for the `corepack`step.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -10,23 +10,26 @@ jobs:
...
 
       - name: Install Dependencies
-        run: yarn
+        run: yarn --immutable
         env:
           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
 
       - name: Create Release Pull Request or Publish to npm
-        uses: changesets/action@v1
+        uses: changesets/action@master
```
Comment (stathismor) [inline .github/workflows/release.yml:32]: question: This means we will be getting the "master" (PC :police_officer: :rotating_light: ) version every time, right? Why was this needed, when all other versions are fixed?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (computerist) [reply .github/workflows/ci.yml:56]: Will discuss on slack - will try removing but I think it is...

File: .github/workflows/release.yml (hunk 4 above)
Comment (computerist) [reply .github/workflows/release.yml:32]: Ah, was trying something. Happy to revert to `@v1`

**Review by stathismor — APPROVED** [review]
